modules/extension/complex/src/main/java/org/geotools/data/complex/expression/FilterFunction_literate.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2019, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.complex.expression;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.LiteralExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.geotools.filter.visitor.DuplicatingFilterVisitor;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.ExpressionVisitor;
+import org.opengis.filter.expression.PropertyName;
+
+public class FilterFunction_literate extends FunctionExpressionImpl {
```

Comment:
@NielsCharlier can you add a comment about what this function does?  From the name, I'm not sure.

Reply:
I hate this class name, the originals are all made this way because they were generated. 

Please rename to LiterateFunction